src/main/java/org/zephyrsoft/sdb2/gui/MainWindow.java

```diff
@@ -621,10 +623,113 @@ private void indexAllSongs() {
 		indexer.index(IndexType.ALL_SONGS, songsModel.getSongs());
 	}
 	
+	@Override
+	public void OnIndexChange() {
+		comboBoxLanguage.removeAllItems();
+		Set<String> languages = new HashSet<>();
+		// Fill in used languages:
+		for (Song song : songsModel.getSongs()) {
+			if (song.getLanguage() != null && !song.getLanguage().isBlank()) {
+				languages.add(song.getLanguage());
+			}
+		}
+		for (String language : languages) {
+			comboBoxLanguage.addItem(language);
+		}
```

Comment:
This should be rewritten as a stream with `.forEach(comboBoxLanguage::addItem)` at the end. The stream can also do the reduction to distinct values.

Reply:
Yes, there are several places where a stream would be a better choice. Right now I committed only the forEach construct.